packages/core/src/lib/plugins/Plugin.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+export type Plugin = {
+  name: string;
+  build?: {
```

Comment:
I think we can make (almost) everything here non-nullable. `repositories`, `dependencies` and `imports` can just default to empty lists, and that in turn means that `build` can also be non-nullable as well (just defaults to the empty  `repositories` and `dependencies`). Then we can remove half of the null checks from `PluginManager`.

We may not be able to do the same with `launchUrl` though.

Reply:
Yeah, moved this to be an interface. Some plugins will require a config, and it's easier like that.